Packs/Bitbucket/Integrations/Bitbucket/Bitbucket.py

```diff
@@ -0,0 +1,1479 @@
+from typing import Dict
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+
+    def __init__(self, workspace: str, server_url: str, auth: tuple, repository: str,
+                 proxy: bool = False, verify: bool = True):
+        self.repository = repository
+        self.workspace = workspace
+        self.server_url = server_url
+        super().__init__(base_url=server_url, auth=auth, proxy=proxy, verify=verify)
+
+    def get_full_url(self, full_url: str) -> Dict:
+        """ Makes a general GET request according to the given full_url.
+        Args:
+            full_url: str - The full url for the GET method.
+        Returns:
+            A dictionary, response object from the given full_url.
+        """
+        return self._http_request(method='GET', full_url=full_url)
+
+    def get_project_list_request(self, page: int | None, page_size: int, project_key: str = None) -> Dict:
+        """ Makes a GET request to Bitbucket in order to get a list of all the project in Bitbucket.
+            If a specific project_key is given, it will return the information about this project.
+        Args:
+            project_key: str - A key to a specific project.
+            page: int - The specific result's page.
+            page_size: int - The number of wanted results in a results page.
+        Returns:
+            A response object in a form of a dictionary.
+        """
+        params = {
+            'page': page,
+            'pagelen': page_size
+        }
+        if not project_key:
+            full_url = f'{self.server_url}/workspaces/{self.workspace}/projects/'
+        else:
+            project_key = project_key.upper()
+            full_url = f'{self.server_url}/workspaces/{self.workspace}/projects/{project_key}'
+        return self._http_request(method='GET', full_url=full_url, params=params)
+
+    def get_open_branch_list_request(self, repo: str, page: int | None, page_size: int) -> Dict:
+        """ Makes a GET request to Bitbucket in order to get a list of all the open branches in Bitbucket.
+        Args:
+            repo: str - The repository the user entered if he did.
+            page: int - The specific result's page.
+            page_size: int - The number of wanted results in a results page.
+        Returns:
+            A response object in a form of a dictionary.
+        """
+        params = {'page': page,
+                  'pagelen': page_size}
+        url_suffix = f'/repositories/{self.workspace}/{repo}/refs/branches'
+        return self._http_request(method='GET', url_suffix=url_suffix, params=params)
+
+    def get_branch_request(self, branch_name: str, repo: str) -> Dict:
+        """ Makes a GET request to Bitbucket in order to get the information of a specific branch in Bitbucket.
```

Comment:
In all the descriptions, no need to specify the method. Just write what the request does.
For example: `Get information on a specific branch`